### ISSUE TITLE
[Upgrades] Enable features when a user Upgrades their workspace to access them

### ISSUE THREAD

**Issue by garrettmknight** [Expensify/App#45692]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** All
**Reproducible in staging?:** Y
**Reproducible in production?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @garrettmknight 
**Slack conversation:** https://expensify.slack.com/archives/C036QM0SLJK/p1721318654213289?thread_ts=1721057410.589669&cid=C036QM0SLJK

## Action Performed:

1. Create a workpace
2. Go to More Features
3. Click/tap on the slider for ‘Report fields’
4. Click ‘Upgrade’
5. Click ‘Got it, thanks’

## Expected Result:
 Report fields is enabled and in the LHN and the user is directed back to the ‘More features’ page

## Actual Result:
User is redirected to Profile page and Report fields isn’t enabled.
## Workaround:

<details><summary>Issue Owner</summary>Current Issue Owner: @garrettmknight</details>

**Comment by garrettmknight** [comment]
Here are the routes that each feature we've just enabled upgrades should take after the user clicks 'Got it, thanks':
- Report fields: More features with Report fields enabled.
- NetSuite: Accounting with RHP for NetSuite setup showing /accounting/netsuite/token-input
- Sage Intacct: Accounting with RHP for Sage Intacct setup showing /accounting/sage-intacct/prerequisites
- GL codes is already good, but for posterity: The path that the user followed to access, either Categories or Tags, with the RHP open to the GL code editor /gl-code
- Payroll codes: Categories with the RHP open to the Payroll code editor /payroll-code
- Tax codes: Tax page with the RHP open to the Tax code /tax-code

**Comment by allroundexperts** [comment]
@garrettmknight Can you clarify on what should be the behaviour if:
a. On the confirmation screen, if the user presses the `back` button instead of the `got it` button.
b. On the confirmation screen, If the user closes the RHP, by clicking outside of it.

**Comment by allroundexperts** [comment]
Draft PR is up btw. Just waiting on the above answers.

**Comment by garrettmknight** [comment]
I think they should both be treated the same.

**Comment by garrettmknight** [comment]
@johncschuster handing this one off while I'm OOO - should be an easy one, fix is already in PR.

**Comment by garrettmknight** [comment]
Thanks for the help @johncschuster - I'll take this one back over.

**Comment by garrettmknight** [comment]
Removing awaiting payment while we await that new PR to get merged and go out.

**Comment by garrettmknight** [comment]
Confirmed this is live and working.
